lib/test-service-provider.js

```diff
@@ -46,42 +46,58 @@
 	 */
 	Box.TestServiceProvider = function(serviceStubs, allowedServicesList) {
 		this.stubs = serviceStubs || {};
-		this.allowedServices = {};
-
-		// Make lookup fast, convert to a map
-		var allowedServices = allowedServicesList || [];
-		for (var i = 0, len = allowedServices.length; i < len; i++) {
-			this.allowedServices[allowedServices[i]] = true;
-		}
+		this.allowedServicesList = allowedServicesList || [];
+		this.serviceInstances = {}; // Stores the instances of pre-registered services
 	};
 
 	Box.TestServiceProvider.prototype = {
 
 		/**
+		 * Check if service is allowed. This should be replaced by Array.prototype.indexOf when we drop IE 8 support
+		 * @param {string} serviceName The name of the service being checked
+		 * @returns {boolean} Returns true if service is in the allowed list
+		 */
+		isServiceAllowed: function(serviceName) {
```

Comment:
I think so, since this method has to do with the internal workings rather than mimicking some feature of Application. At the very least I'd prefix with an underscore as a signal that you shouldn't be using it directly.